### PR TITLE
[JENKINS-71715] Update minimum supported Jenkins version to 2.440 to be able to implement new `ConsoleUrlProvider` API from Jenkins core

### PR DESCRIPTION
**Pull request by dwnusbaum**
See [JENKINS-71715](https://issues.jenkins.io/browse/JENKINS-71715) and https://github.com/jenkinsci/jenkins/pull/8321. (Replaces #288 because I forgot to ask Julien for write access to his branch before he went on vacation.)

### Testing done

See this video which shows me using the new extension with a Pipeline build:

https://github.com/jenkinsci/pipeline-graph-view-plugin/assets/1068968/4713f4d0-16fc-4c4d-837f-e32a362c9a5b

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,7 +45,7 @@
         <!-- Pick up common dependencies for the selected LTS line: https://github.com/jenkinsci/bom#usage -->
         <groupId>io.jenkins.tools.bom</groupId>
         <artifactId>bom-2.361.x</artifactId>
-        <version>1945.v13a_1306588ff</version>
+        <version>1948.veb_1fd345d3a_e</version>
```
Comment (dwnusbaum) [inline pom.xml:48]: We need https://github.com/jenkinsci/bom/pull/1897 for the plugin to start with `hpi:run`.

File: package-lock.json
Comment (dwnusbaum) [inline package-lock.json:1]: Not sure why this got updated

File: src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewUrlProvider.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+public class PipelineConsoleViewUrlProvider implements ConsoleUrlProvider {
+
+    @DataBoundConstructor
+    public PipelineConsoleViewUrlProvider() { }
+
+    @Override
+    public String getConsoleUrl(Run<?, ?> run) {
+        if (run instanceof WorkflowRun) {
+            return run.getUrl() + PipelineConsoleViewAction.URL_NAME;
+        }
```
Comment (dwnusbaum) [inline src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewUrlProvider.java:20]: From a very quick look, I noticed that the visualization in this plugin does not show the overall log in any form, so for example, if your Pipeline has no stages, it shows has a blank page, and things like messages around Jenkins restarting will not be shown, which could be problematic in practice, but at least for now this is just a PoC for the new API.

**Review by timja — APPROVED** [review]
LGTM pending release of https://github.com/jenkinsci/pipeline-input-step-plugin/pull/145, https://github.com/jenkinsci/workflow-job-plugin/pull/373

and bumping jenkins.version to 2.440 and bom

File: src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewUrlProvider.java (hunk 2 above)
Comment (timja) [reply src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewUrlProvider.java:20]: for now https://github.com/jenkinsci/pipeline-graph-view-plugin/pull/289/commits/5e9e5a2fd7d9ece86375eb4badae19dae1b98656 should be enough I think

image

File: src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewUrlProvider.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        return null;
+    }
+
+    @Extension
+    @Symbol("pipelineGraphView")
+    public static class DescriptorImpl extends Descriptor<ConsoleUrlProvider> {
+        @Override
+        public String getDisplayName() {
+            return "Pipeline Graph View"; // TODO: Would it be better to match the action name ("Pipeline Console")?
```
[the remark this answers is not in the document]
Comment (timja) [reply src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewUrlProvider.java:29]: fine for this for now

**Pull request merged**